boot/bootutil/src/bootutil_misc.c

```diff
@@ -320,6 +348,60 @@ boot_read_swap_size(uint32_t *swap_size)
     return rc;
 }
 
+#ifdef MCUBOOT_ENC_IMAGES
+int
+boot_read_enc_key(uint8_t slot, uint8_t *enckey)
+{
+    uint32_t magic[BOOT_MAGIC_SZ];
+    uint32_t off;
+    const struct flash_area *fap;
+    int rc;
+
+    rc = flash_area_open(FLASH_AREA_IMAGE_0, &fap);
```

Comment:
Given the function parameters, should this be flash_area_open(slot,...)?

Reply:
@mkiiskila Nope, the temporary keys are written to either slot0 or scratch (during swap procedure). The slot parameter is here because both keys are saved temporarily and the correct one is indexed by `slot`.